asmcomp/amd64/reload.ml

```diff
@@ -66,7 +67,18 @@ inherit Reloadgen.reload_generic as super
 
 method! reload_operation op arg res =
   match op with
-  | Iintop(Iadd|Isub|Iand|Ior|Ixor|Icomp _|Icheckbound) ->
+  | Iintop(Iadd|Isub|Iand|Ior|Ixor|Icheckbound) ->
+      (* One of the two arguments can reside in the stack, but not both *)
+      if stackp arg.(0) && stackp arg.(1)
+      then ([|arg.(0); self#makereg arg.(1)|], res)
+      else (arg, res)
+  | Iintop(Icomp _) ->
+      (* The result must be a register *)
+      let res =
+        if stackp res.(0)
+        then [|self#makereg res.(0)|]
+        else res
+      in
```

Comment:
`makereg` is a no-op if the argument is already a register, and there's `makeregs` to handle arrays of regs.  Proposed simplification:
```suggestion
      (* The result must be a register (PR#11083) *)
      let res = self#makeregs res in
```

Reply:
I didn't think of calling `makereg` even in the register case, but I didn't use `makeregs` because it wasn't exported. I assume that there's no harm in exporting it though, so I've pushed a patch with your suggestion.